src/common/cuda_utils.h

```diff
@@ -479,6 +479,11 @@ static inline __device__ void atomicAdd(mshadow::half::half_t *address,
   } while (assumed != old);
 }
 
+// Overload atomicAdd to work for signed int64 on all architectures
+static inline  __device__  void atomicAdd(int64_t *address, int64_t val) {
+  atomicAdd(reinterpret_cast<unsigned long long*>(address), static_cast<unsigned long long>(val)); // NOLINT
```

Comment:
are you sure this works for negative value?

Reply:
It should be safe if CUDA uses 2's complement to implement the signed long long.

Reply:
I found this: https://devtalk.nvidia.com/default/topic/506105/atomicadd-with-signed-long-long-not-working/